File: windows/Liferay.SDK/Service/V73/Commercecurrency/CommercecurrencyService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercecurrency
{
	public class CommercecurrencyService : ServiceBase
	{
		public CommercecurrencyService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> SetActiveAsync(long commerceCurrencyId, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/set-active", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceCurrencyAsync(long companyId, string code)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("code", code);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currency", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceCurrencyAsync(long commerceCurrencyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currency", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceCurrencyAsync(long userId, string code, IDictionary<string, string> nameMap, string symbol, IDictionary<string, object> rate, IDictionary<string, string> formatPatternMap, int maxFractionDigits, int minFractionDigits, string roundingMode, bool primary, double priority, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("code", code);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("symbol", symbol);
			_parameters.Add("rate", rate);
			_parameters.Add("formatPatternMap", formatPatternMap);
			_parameters.Add("maxFractionDigits", maxFractionDigits);
			_parameters.Add("minFractionDigits", minFractionDigits);
			_parameters.Add("roundingMode", roundingMode);
			_parameters.Add("primary", primary);
			_parameters.Add("priority", priority);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/add-commerce-currency", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceCurrencyAsync(long commerceCurrencyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/delete-commerce-currency", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchPrimaryCommerceCurrencyAsync(long companyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/fetch-primary-commerce-currency", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceCurrenciesAsync(long companyId, bool active, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("active", active);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currencies", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceCurrenciesAsync(long companyId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currencies", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceCurrenciesCountAsync(long companyId, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currencies-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceCurrenciesCountAsync(long companyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/get-commerce-currencies-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceCurrencyAsync(long commerceCurrencyId, string code, IDictionary<string, string> nameMap, string symbol, IDictionary<string, object> rate, IDictionary<string, string> formatPatternMap, int maxFractionDigits, int minFractionDigits, string roundingMode, bool primary, double priority, bool active, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);
			_parameters.Add("code", code);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("symbol", symbol);
			_parameters.Add("rate", rate);
			_parameters.Add("formatPatternMap", formatPatternMap);
			_parameters.Add("maxFractionDigits", maxFractionDigits);
			_parameters.Add("minFractionDigits", minFractionDigits);
			_parameters.Add("roundingMode", roundingMode);
			_parameters.Add("primary", primary);
			_parameters.Add("priority", priority);
			_parameters.Add("active", active);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/update-commerce-currency", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task UpdateExchangeRateAsync(long commerceCurrencyId, string exchangeRateProviderKey)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);
			_parameters.Add("exchangeRateProviderKey", exchangeRateProviderKey);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/update-exchange-rate", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UpdateExchangeRatesAsync()
		{
			var _parameters = new JsonObject();

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/update-exchange-rates", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> SetPrimaryAsync(long commerceCurrencyId, bool primary)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceCurrencyId", commerceCurrencyId);
			_parameters.Add("primary", primary);

			var _command = new JsonObject()
			{
				{ "/commerce.commercecurrency/set-primary", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercecurrency/CommercecurrencyService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercecurrency;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercecurrencyService extends BaseService {

	public CommercecurrencyService(Session session) {
		super(session);
	}

	public JSONObject addCommerceCurrency(long userId, String code, JSONObject nameMap, String symbol, JSONObject rate, JSONObject formatPatternMap, int maxFractionDigits, int minFractionDigits, String roundingMode, boolean primary, double priority, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("code", checkNull(code));
			_params.put("formatPatternMap", checkNull(formatPatternMap));
			_params.put("maxFractionDigits", maxFractionDigits);
			_params.put("minFractionDigits", minFractionDigits);
			_params.put("nameMap", checkNull(nameMap));
			_params.put("primary", primary);
			_params.put("priority", priority);
			_params.put("rate", checkNull(rate));
			_params.put("roundingMode", checkNull(roundingMode));
			_params.put("symbol", checkNull(symbol));
			_params.put("userId", userId);

			_command.put("/commerce.commercecurrency/add-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceCurrency(long commerceCurrencyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCurrencyId", commerceCurrencyId);

			_command.put("/commerce.commercecurrency/delete-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchPrimaryCommerceCurrency(long companyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);

			_command.put("/commerce.commercecurrency/fetch-primary-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceCurrencies(long companyId, boolean active, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.currency.model.CommerceCurrency>", orderByComparator);

			_command.put("/commerce.commercecurrency/get-commerce-currencies", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceCurrencies(long companyId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.currency.model.CommerceCurrency>", orderByComparator);

			_command.put("/commerce.commercecurrency/get-commerce-currencies", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceCurrenciesCount(long companyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);

			_command.put("/commerce.commercecurrency/get-commerce-currencies-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceCurrenciesCount(long companyId, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("companyId", companyId);

			_command.put("/commerce.commercecurrency/get-commerce-currencies-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject getCommerceCurrency(long commerceCurrencyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCurrencyId", commerceCurrencyId);

			_command.put("/commerce.commercecurrency/get-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCommerceCurrency(long companyId, String code) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("code", checkNull(code));
			_params.put("companyId", companyId);

			_command.put("/commerce.commercecurrency/get-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject setActive(long commerceCurrencyId, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceCurrencyId", commerceCurrencyId);

			_command.put("/commerce.commercecurrency/set-active", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject setPrimary(long commerceCurrencyId, boolean primary) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCurrencyId", commerceCurrencyId);
			_params.put("primary", primary);

			_command.put("/commerce.commercecurrency/set-primary", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceCurrency(long commerceCurrencyId, String code, JSONObject nameMap, String symbol, JSONObject rate, JSONObject formatPatternMap, int maxFractionDigits, int minFractionDigits, String roundingMode, boolean primary, double priority, boolean active, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("code", checkNull(code));
			_params.put("commerceCurrencyId", commerceCurrencyId);
			_params.put("formatPatternMap", checkNull(formatPatternMap));
			_params.put("maxFractionDigits", maxFractionDigits);
			_params.put("minFractionDigits", minFractionDigits);
			_params.put("nameMap", checkNull(nameMap));
			_params.put("primary", primary);
			_params.put("priority", priority);
			_params.put("rate", checkNull(rate));
			_params.put("roundingMode", checkNull(roundingMode));
			_params.put("symbol", checkNull(symbol));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercecurrency/update-commerce-currency", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void updateExchangeRate(long commerceCurrencyId, String exchangeRateProviderKey) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCurrencyId", commerceCurrencyId);
			_params.put("exchangeRateProviderKey", checkNull(exchangeRateProviderKey));

			_command.put("/commerce.commercecurrency/update-exchange-rate", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public void updateExchangeRates() throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_command.put("/commerce.commercecurrency/update-exchange-rates", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

}